docs/advanced-topics/executor-service.md

```diff
@@ -0,0 +1,35 @@
+# Backend ExecutorService
+
+By default JanusGraph uses `ExecutorService` to process some queries in parallel 
+(see `storage.parallel-backend-ops` configuration option).  
+JanusGraph allows to configure the executor service which is used via configuration options 
+provided in `storage.parallel-backend-executor-service` configuration section.
+
+Currently JanusGraph has the next ExecutorService implementations which can be used (controlled via 
+`storage.parallel-backend-executor-service.class`):
+
+* `fixed` - fixed thread pool size;
+* `cached` - cached thread pool size;
+* Custom ExecutorService;
+
+## Custom ExecutorService
+
+To use custom `ExecutorService` the configuration option `storage.parallel-backend-executor-service.class` must be 
+provided which must be the full class name of the `ExecutorService` implementation class.
+
+The provided class which implements `ExecutorService` must have either a public constructor with
+[ExecutorServiceConfiguration](https://javadoc.io/doc/org.janusgraph/janusgraph-core/latest/org/janusgraph/diskstorage/configuration/ExecutorServiceConfiguration.html)
+argument (preferred constructor) or a public parameterless constructor. 
+When both constructors are available the constructor with `ExecutorServiceConfiguration` argument will be used.
+
+When custom `ExecutorService` is provided with `ExecutorServiceConfiguration` it isn't required to 
+use any of the configuration options provided in `ExecutorServiceConfiguration` but sometimes it is
+convenient to use the provided configurations. The configuration options provided in `ExecutorServiceConfiguration` 
+are typically those configuration options which are provided via configurations from `storage.parallel-backend-executor-service`.
+
+# Cassandra backend ExecutorService
+
+Cassandra backend also uses `ExecutorService` to process CQL queries. The rules by which the Cassandra backend 
```

Comment:
(nitpick) I think we can make it clearer by saying sth like "Apart from the general executor service mentioned in the previous section, Cassandra backend uses an additional ExecutorService..."